packages/react-native/src/Authenticator/hooks/useFieldValues/utils.ts

```diff
@@ -176,3 +184,38 @@ export function getRouteTypedFields({
 
   return isVerifyUserRoute ? radioFields : getTypedFields(fields);
 }
+
+/**
+ *
+ * @param field
+ * @param value
+ * @param stateValidations
+ * @returns
+ */
+export const runFieldValidation = (
+  field: TextFieldOptionsType,
+  value: string | undefined,
+  stateValidations: ValidationError | undefined
+): string[] => {
+  let fieldErrors: string[] = [];
+  if (field.required && !value) {
+    fieldErrors.push(getRequiredFieldText());
+  }
+  if (field.type === 'email') {
+    if (!isValidEmail(value)) {
+      fieldErrors.push(getInvalidEmailText());
+    }
+  }
+
+  // add state machine validation errors, if any
+  if (stateValidations && stateValidations[field.name]) {
+    const stateValidation = stateValidations[field.name];
```

Comment:
Think this would be easier to read if `stateValidation` was assigned as a variable outside the `if` statement:
```
const stateFieldValidation = stateFieldValidations?.[field.name];
if (stateFieldValidation) {
  // ...do stuff
}
```